readthedocs-theme/templates/readthedocs/product.html

```diff
@@ -0,0 +1,256 @@
+{% extends "page.html" %}
+
+{% block body_id %}product{% endblock %}
+
+{% block content %}
+
+  <!-- How it works -->
+  <section class="ui vertical segment very padded" id="how-it-works">
+    <div class="ui container center aligned">
+      <h1 class="ui header massive"><<i>missing</i>> Infographic</h1>
+      <p>
+        A very informative and detailed infographic that we still need to draw,<br>explaining the full Read the Docs workings, dependencies and features.
+      </p>
+    </div>
+  </section>
+
+  <!-- User Segments -->
+  <section class="ui vertical segment inverted padded" id="user-segments">
+
+    <div class="ui container">
+      <div class="ui grid stackable three column">
+
+        <div class="column">
+
+          <div>
+
+            <h3 class="ui header inverted"><<i>missing</i>> Illustration</h3>
```

Comment:
I believe it's easier for future iterations to have comments as guidance of what was being done in each section, but since we are not leaving comments anywhere it makes better sense to just remove these too.

Reply:
I would agree that it's help for future authoring purposes to describe the intended future change, but am a little confused by no comments. We do use code comments, however perhaps what hasn't been illustrated is how we try to use code comments?

We generally discard comments that don't provide future code authors with necessary context on the code. We would generally skip code comments that:

* Describe how code operates (`# This is a loop over the project build array`)
* Placeholder comments or code positioning comments
* Commenting out unused code -- we use version control to track old code

In the case, where this block was important to the structure of the design, it might be helpful to keep this in the template. It could also be moved to an issue. A helpful comment would be:

```jinja
{% comment %}
    This is how we will plan on adding illustrations in the future. The content blocks should be broken up by inverted sections:
    <h3 class="ui header inverted"><img class="ui fluid image" src="" /></h3>
{% endcomment %}
```

To illustrate things on the Python side, here's an example -- just the first file I pulled up. The comments here describe to future authors *why* the code is doing what it's doing:
https://github.com/readthedocs/readthedocs.org/blob/master/readthedocs/doc_builder/python_environments.py#L70-L200

Hopefully that helps! This is probably a bit of a code pattern that stems from our Python background and might be a poorly communicated rule because of that.

Reply:
Sorry, I miswrote. I meant to say we are not leaving any commented out code, which, in this case, would function as placeholder for future development.

Knowing the patterns in use with more detail helps. Thanks!

In this case I think it's more helpful to create a separate issue so I opened one here #36.
This allowed to better detail the goals for the illustrations that are needed here and will be a better place for the  discussion on the scope of those images.

Reply:
Ps: We can also refer back to the mockups later, if needed, when adding code back that was already mocked up.